akka-docs/src/main/paradox/coordination.md

```diff
@@ -0,0 +1,113 @@
+not asynchronous so can be called before performing any action for which having the lease is important.
+
+A lease has an owner. If the same owner tries to acquire the lease multiple times it will succeed i.e. leases are reentrant. 
+
+It is important
+to pick a lease name that will be unique for your use case. If a lease needs to be unique for each node in a Cluster the cluster host port can be use:
+
+Scala
+:  @@snip [LeaseDocSpec.scala](/akka-coordination/src/test/scala/docs/akka/coordination/LeaseDocSpec.scala) { #cluster-owner }
+
+Java
+:  @@snip [LeaseDocTest.scala](/akka-coordination/src/test/java/jdocs/akka/coordination/lease/LeaseDocTest.java) { #cluster-owner }
+
+For use cases where multiple different leases on the same node some then something unique to your use cases must be used. For example
+a lease can be used with Cluster Sharding and in this case the shard Id is included in the lease name used for each shard.
+
+## Usages in other Akka modules
+
+Leases can be used for @ref[Cluster Singletons](cluster-singleton.md#lease) and @ref[Cluster Sharding](cluster-sharding.md#lease). 
+
+## Lease implementations
+
+* [Kubernetes API](https://developer.lightbend.com/docs/akka-commercial-addons/current/kubernetes-lease.html)
+
+## Implementing a lease
+
+Implementations should extend
+the @scala[`akka.coordination.lease.scaladsl.Lease`]@java[`akka.coordination.lease.javadsl.Lease`] 
+
+Scala
+:  @@snip [LeaseDocSpec.scala](/akka-coordination/src/test/scala/docs/akka/coordination/LeaseDocSpec.scala) { #lease-example }
+
+Java
+:  @@snip [LeaseDocTest.scala](/akka-coordination/src/test/java/jdocs/akka/coordination/lease/LeaseDocTest.java) { #lease-example }
+
+The methods should provide the following guarantees:
+
+* `acquire` should complete with: `true` if the lease has been acquired, `false` if the lease is taken by another owner, or fail if it can't communicate with the third party system implementing the lease.
+* `release` should complete with: `true` if the lease has definitely been released, `false` if the lease has definitely not been released, or fail if it is unknown if the lease has been released.
+* `checkLease` should return false until an `acquire` Future has completed and should return `false` if the lease is lost due to an error communicating with the third party. Check lease should also not block.
+* The `acquire` lease lost callback should only be called after an `aquire` future has completed and should be called if the lease is lose e.g. due to losing communication with the third party system.
+
+In addition it is expected that if a node that holds a lease that a lease implementation will include a time to live mechanism meaning that it won't be held for ever.
+If a user prefers to have outside intervention in this case for maximum safety then the time to live can be set to infinite.
+
+Create a configuration location with the following configuration:
```

Comment:
it's not good with config (api) that is not defined in reference.conf, perhaps we should define this section in reference.conf after all and use that as a fallback for the user's lease config?

Reply:
Yep, having to define this is a PITA